lambdas/es/indexer/test/test_index.py

```diff
@@ -18,115 +21,206 @@
-            "object": {
-                "key": "hello+world.txt",
-                "size": 100,
-                "eTag": "123456",
-                "sequencer": "0A1B2C3D4E5F678901"
+        "s3SchemaVersion": "1.0"
+    },
+    "userIdentity": {"principalId": "EXAMPLE"}
+}
+
+
+def _check_event(synthetic, organic):
+    # Ensure that synthetic events have the same shape as actual organic ones,
+    # and that overridden properties like bucket, key, eTag are properly set
+    # same keys at top level
+    assert organic.keys() == synthetic.keys()
+    # same value types (values might differ and that's OK)
+    assert {type(v) for v in organic.values()} == \
+        {type(v) for v in synthetic.values()}
+    # same keys and nested under "s3"
+    assert organic["s3"].keys() == synthetic["s3"].keys()
+    assert organic["s3"]["bucket"].keys() == synthetic["s3"]["bucket"].keys()
+    # same value types under S3 (values might differ and that's OK)
+    assert {type(v) for v in organic["s3"].values()} == \
+        {type(v) for v in synthetic["s3"].values()}
+    # spot checks for overridden properties
+    # size absent on delete
+    if "size" in organic["s3"]["bucket"]:
+        assert organic["s3"]["object"]["size"] == synthetic["s3"]["object"]["size"]
+    # versionId absent when unversioned bucket or hard delete
+    if "versionId" in organic["s3"]["bucket"]:
+        assert organic["s3"]["object"]["versionId"] == synthetic["s3"]["object"]["versionId"]
+    # should always be present
+    assert organic["awsRegion"] == synthetic["awsRegion"]
+    assert organic["s3"]["bucket"]["arn"] == synthetic["s3"]["bucket"]["arn"]
+    assert organic["s3"]["bucket"]["name"] == synthetic["s3"]["bucket"]["name"]
+    assert organic["s3"]["object"]["key"] == synthetic["s3"]["object"]["key"]
+    assert organic["s3"]["object"]["eTag"] == synthetic["s3"]["object"]["eTag"]
+
+
+def _make_es_callback(
+        *,
+        errors=False,
+        status=200,
+        unknown_items=False
+):
+    """
+    create a callback that checks the shape of the response
+    TODO: handle errors and delete actions
+    """
+    def check_response(request):
+        raw = [json.loads(line) for line in request.body.splitlines()]
+        # drop the optional source and isolate the actions
+        # see https://www.elastic.co/guide/en/elasticsearch/reference/6.7/docs-bulk.html
+        actions = [line for line in raw if len(line.keys()) == 1]
+        items = [
+            {
+                top_key: {
+                    "_id": values["_id"],
+                    "_index": values["_index"],
+                    "_type": "_doc",
+                    "status": 200
+                }
             }
+            for action in actions
+            for top_key, values in action.items()
+        ]
+        # see https://www.elastic.co/guide/en/elasticsearch/reference/6.7/docs-bulk.html
+        # for response format
+        response = {
+            "took": 5*len(actions),
```

Comment:
You can skip this if you find it less readable.
```suggestion
        items = [
            {
                top_key: {
                    "_id": values["_id"],
                    "_index": values["_index"],
                    "_type": "_doc",
                    "status": 200
                }
            }
            for action in raw
            if len(action) == 1
            for top_key, values in action.items()
        ]
        # see https://www.elastic.co/guide/en/elasticsearch/reference/6.7/docs-bulk.html
        # for response format
        response = {
            "took": 5*len(items),
```